### DISCUSSION TITLE
Enhance search by Postcode (UK)

### DISCUSSION THREAD

**Discussion by derei** [organicmaps/organicmaps#7542] [Ideas]
_This topic is referring to using OrganicMaps in England. I don't know how it behaves in other areas._

Fron what I could observe so far, OrganicMaps is only displaying Postal Boxes when searching by post code. There are few resources available, which, if integrated, could provide a more comprehensive Postcode search. There may be more sources, but these are what I know of:

https://ckan.publishing.service.gov.uk/dataset/ons-postcode-directory-february-2024

https://osdatahub.os.uk/

**Comment by biodranik** [comment]
Thanks! How complete/up-to-date are the datasets? Nominatim uses the os one, right?

**Comment by derei** [comment]
The ONS have been updated February 2024 (last month).
I don't know what Nominatin is using, however ONS and OS perhaps could be used together, complementary. ONS is guaranteed to cover the entire UK and is up to date. OS could provide more localized detailed information.

**Reply by biodranik** [reply]
Did anyone already try to combine/merge both of them for better coverage? It would be great if someone could help with preparing as detailed dataset as possible for the UK and other countries too.

**Comment by hb0nd** [comment]
see https://github.com/organicmaps/organicmaps/issues/717

**Comment by aliomarmohamed** [comment]
I have no experience in open source contributions can I join to organicMaps team to contribute and what is your advice to get started to open source projects ? thanks in advance
